tests/e2e/specs/wcpay/shopper/shopper-checkout-purchase-with-upe-methods.spec.js

```diff
@@ -22,118 +22,158 @@ const card = config.get( 'cards.basic' );
-			await setupProductCheckout(
-				config.get( 'addresses.upe-customer.billing.be' )
-			);
-			page.waitFor( 1000 );
-			await selectOnCheckout( 'bancontact', page );
-			await shopper.placeOrder();
-			await completeRedirectedPayment( page, 'success' );
-			await page.waitForNavigation( {
-				waitUntil: 'networkidle0',
-			} );
-			await expect( page ).toMatch( 'Order received' );
-		} );
-	} );
-
-	describe( 'My Account', () => {
-		let timeAdded;
-
-		it( 'should add the card as a new payment method', async () => {
-			await shopperWCP.goToPaymentMethods();
-			await shopperWCP.addNewPaymentMethod( 'basic', card );
-
-			// Take note of the time when we added this card
-			timeAdded = Date.now();
+const cardTestingPreventionStates = [
+	{ cardTestingPreventionEnabled: false },
+	{ cardTestingPreventionEnabled: true },
+];
 
-			// Verify that the card was added
-			await expect( page ).not.toMatch(
-				'You cannot add a new payment method so soon after the previous one. Please wait for 20 seconds.'
-			);
-			await expect( page ).toMatch( 'Payment method successfully added' );
-			await expect( page ).toMatch(
-				`${ card.expires.month }/${ card.expires.year }`
+describe.each( cardTestingPreventionStates )(
+	'Enabled UPE with deferred intent creation',
+	( { cardTestingPreventionEnabled } ) => {
+		beforeAll( async () => {
+			await merchant.login();
+			await merchantWCP.enablePaymentMethod( UPE_METHOD_CHECKBOXES );
+			if ( cardTestingPreventionEnabled ) {
+				await merchantWCP.enableCardTestingProtection();
+			}
+			await merchant.logout();
+			await shopper.login();
+			await shopperWCP.changeAccountCurrencyTo(
+				config.get( 'addresses.upe-customer.billing.de' ),
+				'EUR'
 			);
-			await waitTwentySecondsSinceLastCardAdded();
+			await shopperWCP.emptyCart();
 		} );
 
-		it( 'should be able to set payment method as default', async () => {
-			await shopperWCP.goToPaymentMethods();
-			await shopperWCP.addNewPaymentMethod( 'basic2', card2 );
-			// Take note of the time when we added this card
-			timeAdded = Date.now();
-
-			// Verify that the card was added
-			await expect( page ).not.toMatch(
-				'You cannot add a new payment method so soon after the previous one. Please wait for 20 seconds.'
-			);
-			await expect( page ).toMatch( 'Payment method successfully added' );
-			await expect( page ).toMatch(
-				`${ card2.expires.month }/${ card2.expires.year }`
-			);
-			await shopperWCP.setDefaultPaymentMethod( card2.label );
-			// Verify that the card was set as default
-			await expect( page ).toMatch(
-				'This payment method was successfully set as your default.'
-			);
+		afterAll( async () => {
+			await shopperWCP.logout();
+			await merchant.login();
+			await merchantWCP.disablePaymentMethod( UPE_METHOD_CHECKBOXES );
+			if ( cardTestingPreventionEnabled ) {
+				await merchantWCP.disableCardTestingProtection();
+			}
+			await merchant.logout();
```

Comment:
Can we add shopperWCP.emptyCart() on afterAll since the test adds products to cart.

In case the test breaks, it might cause a cascade failure.

Reply:
Added it in https://github.com/Automattic/woocommerce-payments/pull/8164/commits/a07d017aa9c3852e1b23538fece0319c4c24b285.